### DISCUSSION TITLE
How can I custom the folder where the serverless functions locates?

### DISCUSSION THREAD

**Discussion by Char2sGu** [vercel/vercel#7591] [Help] [answered]
Next.js projects can place their serverless functions inside the /pages/api folder, but I searched the documents for hours and still can't find a way for projects using other frameworks to locate the functions in folders other than /api.

Can anybody help? Thanks!

**Comment by ryanmstokes** [comment]
I second this question. is this possible somehow?

I want to rewrite /pages/api to whateverishouldwant/api

**Reply by amyegan** [reply]
We do have a way to rewrite urls without modifying the visible url. This would allow you to give the appearance of a `whateverishouldwant/api` path, but you would still need to put the serverless functions in `/pages/api` or `/api` for them to be picked up by Vercel. More info here - https://vercel.com/docs/project-configuration#project-configuration/rewrites

**Comment by AaronBeaudoin** [comment]
I third this question. I have a special function that I'd like to put somewhere else in my project. I'd also like the deployed path of the function to not necessarily be identical to it's path in my project.

For example, perhaps I'd like a function at `renderer/vercel.ts` in my project to be deployed at `<url>/renderer`.

**Comment by amyegan** [comment, marked as answer]
We don't currently have a way to configure a custom directory for serverless functions outside of `/api`. I logged this idea for the team to look into.

**Reply by lissy93** [reply]
Hey @amyegan 
Thanks for your answer, but if it's not possible to put the serverless functions in an alternate directory, why does the [docs give an example](https://vercel.com/docs/projects/project-configuration#functions) of customizing functions within the src/pages/ dir? Or am I totally missing something here?
